nixos/modules/services/torrent/transmission.nix

```diff
@@ -286,6 +309,13 @@ in
           ${pkgs.jq}/bin/jq --slurp add ${settingsFile} '${cfg.credentialsFile}' |
           install -D -m 600 -o '${cfg.user}' -g '${cfg.group}' /dev/stdin \
            '${cfg.home}/${settingsDir}/settings.json'
+          ${lib.optionalString (cfg.dhtBootstrapNodes != [ ]) ''
```

Comment:
I recommend to do something like this
```
let
  bootstrapFile = pkgs.writeText "dht.bootstrap"
    (lib.concatMapStrings (node: "${node.host} ${toString node.port}\n") cfg.dhtBootstrapNodes)
in
''
  ln -sf ${bootstrapFile} ${cfg.home}/${settingsDir}
''
```
over using a here-doc string. This pattern tends to break in subtle ways due to parameter expansions.